server/plugins/openAiDallE3Plugin.js

```diff
@@ -0,0 +1,105 @@
+import RequestDurationEstimator from '../../lib/requestDurationEstimator.js';
+import pubsub from '../pubsub.js';
+import ModelPlugin from './modelPlugin.js';
+
+const requestDurationEstimator = new RequestDurationEstimator(10);
+
+/**
+ * @description This plugin is for the OpenAI DALL-E 3 model.
+ */
+class OpenAIDallE3Plugin extends ModelPlugin {
+    constructor(config, pathway, modelName, model) {
+        super(config, pathway, modelName, model);
+    }
+
+    /**
+     * @description At the time of writing, the DALL-E 3 API on Azure is sync-only, so to support async
+     * we keep the request open and send progress updates to the client 
+     * over a websocket.
+     */
+    async execute(text, parameters, _, pathwayResolver) {
+        const url = this.requestUrl(text);
+        const data = JSON.stringify({ prompt: text });
+
+        let id;
```

Comment:
Unused?

Reply:
This is unused, an artifact of the copy-paste from the DALL-E 2 plugin. Intended to use it but surprisingly, DALL-E 3 actually does *not* provide us an ID back, at least not at the moment on Azure.